app/components/avo/actions_component.rb

```diff
@@ -2,40 +2,47 @@
 
 class Avo::ActionsComponent < Avo::BaseComponent
   include Avo::ApplicationHelper
-  attr_reader :label, :size, :as_row_control
-
-  def initialize(actions: [], resource: nil, view: nil, exclude: [], include: [], style: :outline, color: :primary, label: nil, size: :md, as_row_control: false, icon: nil)
-    @actions = actions || []
-    @resource = resource
-    @view = view
-    @exclude = Array(exclude)
-    @include = include
-    @color = color
-    @style = style
-    @label = label || I18n.t("avo.actions")
-    @size = size
-    @icon = icon
-    @as_row_control = as_row_control
+
+  ACTION_FILTER = _Set(_Class(Avo::BaseAction))
+
+  prop :actions, _Array(Avo::BaseAction), default: -> { [] }
+  prop :resource, _Nilable(Avo::BaseResource)
+  prop :view, _Nilable(Symbol), &:to_sym
+  prop :exclude, _Nilable(ACTION_FILTER), &:to_set
+  prop :include, _Nilable(ACTION_FILTER), &:to_set
```

Comment:
Couple of points on these: I updated them to be Sets since we do repeated _inclusion_ lookups. Arrays will still be accepted because of the `&:to_set` coercion.

One thing I haven’t carried over, which you may need me to bring back is the `Array()` coercion that used to be on the exclude list but not on the include list. This would have allowed you to pass a single non-enumerable object and have it wrapped in an Array. If we bring that back, we should probably add it to the include list as well.

Reply:
Both should allow a single non-enumerable object or an array of objects